### PR TITLE
Wisconsin Regional Changes

### PR DESCRIPTION
**Pull request by rguinn2338**
Merge in changes from Wisconsin Regional

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -126,16 +127,16 @@
...
-    public static final double COMPETITION_FRONT_RIGHT_MODULE_STEER_OFFSET = -Math.toRadians(352.2);
+    public static final double COMPETITION_FRONT_RIGHT_MODULE_STEER_OFFSET = -Math.toRadians(352.2-.8);
 
     public static final double PRACTICE_BACK_LEFT_MODULE_STEER_OFFSET = -Math.toRadians(221.4019);
-    public static final double COMPETITION_BACK_LEFT_MODULE_STEER_OFFSET = -Math.toRadians(0.39);
+    public static final double COMPETITION_BACK_LEFT_MODULE_STEER_OFFSET = -Math.toRadians(0.39-.34);
 
     public static final double PRACTICE_BACK_RIGHT_MODULE_STEER_OFFSET = -Math.toRadians(244.2654);
-    public static final double COMPETITION_BACK_RIGHT_MODULE_STEER_OFFSET = -Math.toRadians(-41.36);
+    public static final double COMPETITION_BACK_RIGHT_MODULE_STEER_OFFSET = -Math.toRadians(-41.36+.7);
```
Comment (duffym22) [inline src/main/java/frc/robot/Constants.java:139]: Condense offsets

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -90,7 +91,7 @@
   private JoystickButton operatorEjectButton;
   private JoystickButton operatorFeedButton;
   private JoystickButton operatorShooterOnButton;
-  private JoystickButton operatorHoodSwitch;
+  //private JoystickButton operatorHoodSwitch;
```
Comment (duffym22) [inline src/main/java/frc/robot/RobotContainer.java:94]: Remove if no longer used

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -116,6 +117,7 @@ public RobotContainer(RobotConfig rc) {
   DoubleSupplier m_translationXSupplier = new DoubleSupplier() {
     @Override
     public double getAsDouble() {
+      var robotState = drivetrainSubsystem.getGyroscopeRotation();
```
Comment (duffym22) [inline src/main/java/frc/robot/RobotContainer.java:120]: robotState is not used here - remove if not needed

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -293,8 +297,8 @@ private void configureButtonBindings() {
       driverStopShooter.whenPressed(new StopShooterCommand(shooter));
       operatorShooterOnButton.whenHeld(new RunShooterCommand(shooter))
           .whenReleased(new StopShooterCommand(shooter));
-      operatorHoodSwitch.whenActive(new InstantCommand(() -> shooter.extendHood()));
-      operatorHoodSwitch.whenInactive(new InstantCommand(() -> shooter.retractHood()));
+      //operatorHoodSwitch.whenActive(new InstantCommand(() -> shooter.extendHood()));
+      //operatorHoodSwitch.whenInactive(new InstantCommand(() -> shooter.retractHood()));
```
Comment (duffym22) [inline src/main/java/frc/robot/RobotContainer.java:301]: Remove if no longer used

File: src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java [hunk 5]
```diff
@@ -122,13 +122,14 @@ private static Rotation2d getTrajectoryRotation(List<Translation2d> waypoints) {
...
     protected static final Pose2d WAYPOINT_B = new Pose2d(0.82910 * scaler, 0.661008 * scaler, Rotation2d.fromDegrees(0));
     protected static final Pose2d WAYPOINT_C = new Pose2d(1.653747 - 0.2 * scaler, 2.67440 + 0.6* scaler, Rotation2d.fromDegrees(-112));
-    protected static final Pose2d WAYPOINT_D = new Pose2d(2.3598 - 0.6 * scaler, 1.5827 + 0.6* scaler, Rotation2d.fromDegrees(-42));
+    protected static final Pose2d WAYPOINT_D = new Pose2d(2.3598 - 0.6 * scaler, 1.5827 + 0.6* scaler, Rotation2d.fromDegrees(-42 - 8));
     protected static final Pose2d WAYPOINT_E = new Pose2d(1.51203 - 0.2 * scaler, 6.7617 + 0.5 * scaler, Rotation2d.fromDegrees(-45)); //0.8
     protected static final Pose2d WAYPOINT_F = new Pose2d(5.16403 * scaler, 2.18839 * scaler, Rotation2d.fromDegrees(-135));
     protected static final Pose2d WAYPOINT_G = new Pose2d(5.82539 +  0.3 * scaler, 2.8711 + 0.3 * scaler, Rotation2d.fromDegrees(-135));
-    protected static final Pose2d WAYPOINT_H = new Pose2d(1.7778 * scaler, 0 * scaler, Rotation2d.fromDegrees(0));
+    protected static final Pose2d WAYPOINT_H = new Pose2d(1.7778 * scaler, 0 * scaler, Rotation2d.fromDegrees(0-2));
```
Comment (duffym22) [inline src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java:129]: Condense

File: src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java [hunk 6]
```diff
@@ -122,13 +122,14 @@ private static Rotation2d getTrajectoryRotation(List<Translation2d> waypoints) {
     protected static final Pose2d WAYPOINT_A = new Pose2d(2.9238 * scaler, 0.41186 * scaler, Rotation2d.fromDegrees(-21));
     protected static final Pose2d WAYPOINT_B = new Pose2d(0.82910 * scaler, 0.661008 * scaler, Rotation2d.fromDegrees(0));
     protected static final Pose2d WAYPOINT_C = new Pose2d(1.653747 - 0.2 * scaler, 2.67440 + 0.6* scaler, Rotation2d.fromDegrees(-112));
-    protected static final Pose2d WAYPOINT_D = new Pose2d(2.3598 - 0.6 * scaler, 1.5827 + 0.6* scaler, Rotation2d.fromDegrees(-42));
+    protected static final Pose2d WAYPOINT_D = new Pose2d(2.3598 - 0.6 * scaler, 1.5827 + 0.6* scaler, Rotation2d.fromDegrees(-42 - 8));
```
Comment (duffym22) [inline src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java:125]: Condense

File: src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java [hunk 7]
```diff
@@ -122,13 +122,14 @@ private static Rotation2d getTrajectoryRotation(List<Translation2d> waypoints) {
...
     protected static final Pose2d WAYPOINT_F = new Pose2d(5.16403 * scaler, 2.18839 * scaler, Rotation2d.fromDegrees(-135));
     protected static final Pose2d WAYPOINT_G = new Pose2d(5.82539 +  0.3 * scaler, 2.8711 + 0.3 * scaler, Rotation2d.fromDegrees(-135));
-    protected static final Pose2d WAYPOINT_H = new Pose2d(1.7778 * scaler, 0 * scaler, Rotation2d.fromDegrees(0));
+    protected static final Pose2d WAYPOINT_H = new Pose2d(1.7778 * scaler, 0 * scaler, Rotation2d.fromDegrees(0-2));
     protected static final Pose2d WAYPOINT_I = new Pose2d(4.5541 * scaler, 1.21188 * scaler, Rotation2d.fromDegrees(-111));
     protected static final Pose2d WAYPOINT_J = new Pose2d(1.8958 * scaler, 0.8403 * scaler, Rotation2d.fromDegrees(-21));
     protected static final Pose2d WAYPOINT_K = new Pose2d(0.651352 * scaler, 1.2713 * scaler, Rotation2d.fromDegrees(-21));
     protected static final Pose2d WAYPOINT_L = new Pose2d(0.634897 * scaler, 0, Rotation2d.fromDegrees(0));
+    protected static final Pose2d WAYPOINT_M = new Pose2d(2.3598 - 0.6 * scaler, 1.5827 + 0.6 + 0.8 * scaler, Rotation2d.fromDegrees(-55));
```
Comment (duffym22) [inline src/main/java/frc/robot/commands/auto/DriveSegmentBaseCommand.java:134]: Condense

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 8]
```diff
@@ -199,10 +199,10 @@ public void setRpmTargetUsingDistance(double distance) {
       extendHood();
     }
     if (_hoodPosition) {
-      double speed = 14.2857 * distance * distance - 221.4286 * distance + 2807.1429;
+      double speed = 4.120879 * distance * distance - 11.263736 * distance + 1741.758242+20;
       setTargetNearRpm(speed);
     } else {
-      double speed = 12.5 * distance * distance - 125 * distance + 2212.5;
+      double speed = 9.375 * distance * distance - 50 * distance + 1940.625;
```
Comment (duffym22) [inline src/main/java/frc/robot/subsystems/Shooter.java:205]: Do these numbers need to be constants?

**Review by borkae — APPROVED** [review]

**Comment by borkae** [comment]
What Matt said.

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 8 above)
Comment (rguinn2338) [reply src/main/java/frc/robot/subsystems/Shooter.java:205]: These are our shooter quadratics, they are generated from an external app @dkScammer wrote.
We should integrate that app into our code and use constants for the inputs to it

**Pull request merged**